frontend/test/pytest/test_gradient.py

```diff
@@ -1780,5 +1780,35 @@ def fn(x):
         assert np.allclose(res_pattern_partial, expected)
 
 
+@pytest.mark.parametrize("argnums", [0, 1, (0, 1)])
+@pytest.mark.parametrize("transform_qjit", [False, True])
+def test_grad_argnums(argnums, transform_qjit):
+    """Tests https://github.com/PennyLaneAI/catalyst/issues/1477"""
+    @qml.qnode(device=qml.device("lightning.qubit", wires=4), interface="jax")
+    def circuit(inputs, weights):
+        qml.AngleEmbedding(features=inputs, wires=range(4), rotation="X")
+        for i in range(1, 4):
+            qml.CRX(weights[i - 1], wires=[i, 0])
+        return qml.expval(qml.PauliZ(wires=0))
+
+    if transform_qjit:
+        circuit = qjit(circuit)
```

Comment:
I don't think we need to test without qjit, since we would be testing PennyLane in that case (which has its own test suite):
```suggestion
def test_grad_argnums(argnums):
    """Tests https://github.com/PennyLaneAI/catalyst/issues/1477"""
    @qjit
    @qml.qnode(device=qml.device("lightning.qubit", wires=4), interface="jax")
    def circuit(inputs, weights):
        qml.AngleEmbedding(features=inputs, wires=range(4), rotation="X")
        for i in range(1, 4):
            qml.CRX(weights[i - 1], wires=[i, 0])
        return qml.expval(qml.PauliZ(wires=0))
```